### PR TITLE
Feature/91 add glossary to client

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [EQ-91](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-91)

## Main Changes:
* Added the [glossary component](https://github.com/Eastern-Research-Group/glossary) to the client-side UI.
* Added two _eslint_-related dev dependencies to the server.
  * The two plugins were specified in the server-side config, but not yet installed.

## Steps To Test:
1. Navigate to the application's home page: [localhost:3000/attains](localhost:3000/attains).
2. Click the **Glossary** button and confirm the glossary slides open from the right.
3. Alternatively, a test glossary link has been added to the "Current Query" label. When clicked, this should link to the "Acidity" glossary term.

### CODE REVIEW

File: app/client/src/routes/home.tsx [hunk 1]
```diff
@@ -348,109 +358,138 @@ export function Home() {
...
+                    className="margin-x-auto usa-button usa-button--outline"
+                    onClick={(_ev) => inputDispatch({ type: 'reset' })}
+                    type="button"
+                  >
+                    Clear Search
+                  </button>
+                </div>
+              </div>
+              <h4>
+                <GlossaryTerm term="Acidity">Current Query</GlossaryTerm>
```
Comment (cschwinderg) [inline app/client/src/routes/home.tsx:470]:
```suggestion
                {/* TODO - Remove the glossary linkage before production deployment */}
                <GlossaryTerm term="Acidity">Current Query</GlossaryTerm>
```

Can you add a TODO comment here? I just don't want this example glossary linkage to make it to production.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Works well and looks good! 

I just have some small suggestions/comments below.

File: app/client/src/routes/home.tsx [hunk 2]
```diff
@@ -98,8 +101,9 @@ function getInputValue(input: Exclude<InputState[keyof InputState], null>) {
 }
 
 // Uses URL query parameters or default values for initial state
+// eslint-disable-next-line
 async function getUrlInputs(
-  signal: AbortSignal, // will need for DB checks
+  _signal: AbortSignal,
   domainOptions: DomainValues,
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/routes/home.tsx:107]: Yes, good call.

File: app/client/src/routes/home.tsx [hunk 3]
```diff
@@ -98,8 +101,9 @@ function getInputValue(input: Exclude<InputState[keyof InputState], null>) {
 }
 
 // Uses URL query parameters or default values for initial state
+// eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/routes/home.tsx:104]: I was having issues with my configuration for a bit, but it's back in line now so I'll remove that. There are a couple lines in the `custom.d.ts` file, too, so I'll remove those.

[the remark this answers is not in the document]
Comment (cschwinderg) [reply app/client/src/routes/home.tsx]: Makes sense. Thank you for removing those.

**Review by cschwinderg — APPROVED** [review]
Thanks for making those changes.

**Pull request merged**